viewing_party/party.py

```diff
@@ -1,23 +1,471 @@
+#     # This represents that the user has a list of watched movies
+#     watched = user_data.get("watched", [])
+#     # The average rating of an empty watched list is `0.0`
+#     if not watched:
+#         return 0.0
+#     # Calculate the average rating of all movies in the watched list
+#     total_ratings = sum(movie.get("rating", 0.0) for movie in watched)
+#     # return the average rating
+#     avg_rating = total_ratings / len(watched)
+#     return avg_rating
+
+# ========================================= wave 02- # 2. get_most_watched_genre ========== RC
+
+def get_most_watched_genre(user_data):
+    # keep the var the same
+    watched = user_data["watched"]
+    # if movie title not watched, return none
+    if not watched:
+        return None
+    
+    # iterate thru each "genre" key's value for the entire length of nested dictionary
+    genres_watched = [watched[i]["genre"] for i in range(len(watched))]
+    
+    # get the most watched genre with max function
+    return max(genres_watched, key=genres_watched.count)
+
+# ========================================= wave 02- # 1. get_watched_avg_rating ========== RC
+
+# user_data = {
+#     'watched': [
+#         {'genre': 'Fantasy', 'rating': 4.8, 'title': 'The Lord of the Functions: The Fellowship of the Function'}... 'rating': 2.0, 'title': 'Recursion'}, 
+#         {'genre': 'Intrigue', 'rating': 4.5, 'title': 'Instructor Student TA Manager'}]}
+
+# def get_watched_avg_rating(user_data):
+#     # This represents that the user has a list of watched movies
+#     watched = user_data.get("watched", [])
+    
+#     # The average rating of an empty watched list is `0.0`
+#     if not watched:
+#         return 0.0
+    
+#     # Calculate the average rating of all movies in the watched list
+#     total_ratings = sum(movie.get("rating", 0.0) for movie in watched)
+    
+#     # return the average rating
+#     avg_rating = total_ratings / len(watched)
+#     return avg_rating
+
+# ========================================= wave 02- # 2. get_most_watched_genre ========== RC
+
+# def get_most_watched_genre(user_data):
+#     watched_genres = {}
+
+# # - Determine which genre is most frequently occurring in the watched list
+#     # iterate through 
+#     for movie in user_data["watched"]:
+#         genre = movie["genre"]
+#         if genre in watched_genres:
+#             watched_genres["genre"] += 1
+#         else:
+#             watched_genres["genre"] = 1
+
+#     if not watched_genres:
+#         return None 
+#     print(watched_genres)
+#     return max(watched_genres, key=watched_genres.get)
+
+
+# ========================================= wave 02- # 2. get_watched_avg_rating ========== SJ
+
+def get_watched_avg_rating(user_data): 
+    
+    #  initialize a variable for ratings  set to 0     
+    ratings_total = 0
+    average_rating = 0
+
+    #  if the length of the user_ data( "watched") is empty  
+    if len(user_data["watched"]) == 0: 

```

Comment:
Please refactor this using the “pythonic” way to check for falsy values in Python.